### ISSUE TITLE
Failures when running gradle locally

### ISSUE THREAD

**Issue by bendk** [mozilla/application-services#4054]
The `./gradlew test` is failing for me when I run it locally.  The failing tasks are `:crashtest:testDebugUnitTest` and `:crashtest:testReleaseUnitTest` .  I believe markh said that the tests are failing for him as well.  We don't run those tests in CI.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/SYNC-2174)
┆Story Points: 2
┆Epic: local development, ci and release (backlog)
┆Sprint: SYNC - 2021-06-11

**Comment by bendk** [comment]
I just looked through the `.cargo/config.yml` and I don't see the gradle tests being run there either.  So maybe it's not a local issue, maybe the tests aren't passing in general.

**Comment by rfk** [comment]
FWIW I can also reproduce these failures locally.

> I just looked through the .cargo/config.yml and I don't see the gradle tests being run there either. 

IIRC the gradle/android tests get run in taskcluster (from under `./taskcluster/`) rather than in Circle (which I assume it what you meant here with the `.yml` file).

**Comment by rfk** [comment]
Hrm, doing a `./gradlew crashtest:clean` seems to have fixed the issue for me. @bendk can you confirm whether this is still failing for you locally, even after a clean?

**Comment by rfk** [comment]
The errors @bendk is getting are different from mine:

```
> Task :crashtest:testDebugUnitTest FAILED
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.robolectric.util.ReflectionHelpers$7 (file:/home/bdk/.gradle/caches/modules-2/files-2.1/org.robolectric/shadowapi/3.8/d638f001e81ef737bb35db2964312360cc996a94/shadowapi-3.8.jar) to method java.lang.ClassLoader.getPackage(java.lang.String)
WARNING: Please consider reporting this to the maintainers of org.robolectric.util.ReflectionHelpers$7
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
mozilla.appservices.crashtest.CrashTestHelpersTest > testErrorsAreThrown FAILED
    java.lang.NoClassDefFoundError at Class.java:-2
        Caused by: java.lang.ClassNotFoundException at SandboxClassLoader.java:132
            Caused by: java.lang.RuntimeException at ClassInstrumentor.java:125
                Caused by: java.lang.IllegalArgumentException at null:-1
mozilla.appservices.crashtest.CrashTestHelpersTest > testPanicsAreCaughtAndThrown FAILED
    java.lang.NoClassDefFoundError at Class.java:-2
        Caused by: java.lang.ClassNotFoundException at SandboxClassLoader.java:132
            Caused by: java.lang.RuntimeException at ClassInstrumentor.java:125
                Caused by: java.lang.IllegalArgumentException at null:-1
2 tests completed, 2 failed
```

The internet suggests that this error may by using an incompatible Java version, e.g.

* https://github.com/robolectric/robolectric/issues/4776
* https://stackoverflow.com/questions/52761004/robolectric-error-when-running-gradlew-test

@bendk just going on what's suggested in that stackoverflow q, what does `java -version` report for you, and does `./libs/verify-android-environment.sh` execute successfully?

**Comment by bendk** [comment]
Installing Java 1.8 seems to have fixed that error.  On fedora I needed the  `java-1.8.0-openjdk` and  `java-1.8.0-openjdk-devel` packages, then I needed to switch my defaults to use it.

Maybe we could put a version check in `gradlew` similar to the one in `./libs/verify-android-environment.sh`

However, after running that I now get errors in `:nimbus:testDebugUnitTest`: [gradle-report.txt](https://github.com/mozilla/application-services/files/6599651/gradle-report.txt)

**Comment by rfk** [comment]
From the linked report, looks like a bunch of:

```
org.mozilla.experiments.nimbus.NimbusTest > getAvailableExperiments does not return experiments that don't match the appName FAILED
    java.lang.UnsatisfiedLinkError at NimbusTest.kt:58
```

Which looks similar to what I was getting in a comment earlier above, and which I'm confident will be fixed by #4180 which just landed a moment ago.

Optimistically, I'm going to go ahead and close this issue, but we can open a fresh one if we see fresh errors.
